include/vcpkg/binarycaching.h

```diff
@@ -42,6 +48,51 @@ namespace vcpkg
         const IBinaryProvider* m_available_provider = nullptr; // meaningful iff m_status == available
     };
 
+    struct BinaryPackageInformation
```

Comment:
Hmmm I wonder if this scheme makes it easier to pass around shared cache information, like the zip, between config providers so we can get out of the business of multiple destinations trying to share the same provider?

Reply:
Yeah this is the idea behind this PR and implemented. The `IObjectProvider`'s only upload and download objects (in this case the zips, but they also can be used for asset caching). Since the zipping is now not done inside the `IObjectProvider`'s they now represent one destination.